src/Unused/Cache/Fingerprint.hs

```diff
@@ -0,0 +1,25 @@
+module Unused.Cache.Fingerprint
+    ( sha
+    ) where
+
+import System.Process
+
+sha :: IO String
+sha =
+    getSha <$> (fileList >>= md5Input >>= sortInput >>= md5Result)
+  where
+    getSha = head' . lines
+    head' (x:_) = x
+    head' _ = ""
+
+fileList :: IO String
+fileList = readProcess "find" [".", "-type", "f", "-not", "-path", "*/tmp/unused/*", "-print0"] ""
+
+md5Input :: String -> IO String
+md5Input = readProcess "xargs" ["-0", "-P", "1", "md5"]
```

Comment:
Random shell nit, but `find | xargs` can usually be replaced by `find -exec {} +`. Then you don't need the `-print0 | -0` stuff since `find` is executing on the file arguments directly.

```sh
find . -type f -not -path '*/tmp/unused/*' -exec md5 {} +
```

That said, wouldn't it be more fun to write this stuff in Haskell, instead of shelling out? ;)